### ISSUE TITLE
Add a deepmass implementation using the same architecture as our denoiser

### ISSUE THREAD

**Issue by EiffL** [CosmoStat/jax-lensing#14]
This issue is to try to reimplement DeepMass in this framework, to allow for comparison with the previously published method.

This would also allow us to compare mean posteriors with DeepMass predictions, and possibly also check @NiallJeffrey's moments networks.

TODO:
 - [x] : Decide if we want to keep a Wiener filter step (I'd say meh)
 - [x] : Add a script to train a DeepMass model (can use same UNet model as DSM)

**Comment by EiffL** [comment]
@NiallJeffrey would you like to take this on? Otherwise we are happy to do it :-)

**Comment by EiffL** [comment]
@NiallJeffrey , @b-remy is making very good progress on using @liuxx479 's new KappaTNG simulations, so I think we could be able to get our main results for the paper within a couple of weeks at this point.

So, we wanted to run a comparison to DeepMass, and potentially your moments networks. Would you be interested in looking into this? Or should we go ahead and make a deepmass-like model on our own?

It might be slightly still subject to change, but we have a U-Net model that we use as our score network, I think it can be used as-is for deepmass. Only question is whether we want to include the Wiener-filter step or not....

**Comment by NiallJeffrey** [comment]
Did you make a DeepMass model of you're own? If not, I can adapt the existing one. I wanted to chat with Ben regarding Moment Networks  for mass mapping, but he's taking holiday at the moment, so I haven't been able to

**Comment by NiallJeffrey** [comment]
In the DeepMass paper I did it in advance, but one could do it on the fly. I'll look at the new implementation

**Comment by EiffL** [comment]
@b-remy I can't find a mask file for our new dataset, do you have an equivalent of https://github.com/CosmoStat/jax-lensing/blob/master/data/cosmos_mask_0.4arcmin.fits ?

**Comment by b-remy** [comment]
Indeed, there was no mask up to date. There is one here now: https://github.com/CosmoStat/jax-lensing/tree/comos-data/data/COSMOS

**Comment by EiffL** [comment]
I think this is done?

**Comment by EiffL** [comment]
But @b-remy could you push the latest version of your code to the repo?

**Comment by b-remy** [comment]
Hi, I just pushed the updated code for DeepMass. The U-net architecture is the same we used to train our denoiser, on which I removed the division by the noise level (which is zero here). 

Training script can be found [here](https://github.com/CosmoStat/jax-lensing/blob/deepmass/scripts/train_deepmass.py).

and notebook result [here](https://github.com/CosmoStat/jax-lensing/blob/deepmass/notebooks/Test-DeepMass.ipynb).

![image](https://user-images.githubusercontent.com/30293694/126049500-4045cad1-4cb2-424a-9e99-827b0bca0496.png)
